src/main/java/org/springframework/data/release/infra/LicenseHeaderCommands.java

```diff
@@ -77,6 +79,28 @@ public class LicenseHeaderCommands extends TimedCommand {
 
 	List<String> filePatterns = Arrays.asList("pom.xml", "**/*.java", "**/*.kt", "**/*.adoc");
 
+	enum FileType {
+
+		POM_XML, JAVA, KOTLIN, ADOC;
+
+		static FileType of(File file) {
+			var fileName = file.getName().toLowerCase(Locale.ROOT);
```

Comment:
Switch w/ pattern matching would have been nice here but it is a preview feature in 17 and I am not sure what the minimum Java version is expected here.